src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -235,15 +237,34 @@ public boolean getPersonalScan() {
     public DescriptorImpl getDescriptor() {
         return (DescriptorImpl)super.getDescriptor();
     }
-    
-    @Override
-    public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
-	perform((Run<?,?>)build, launcher, listener);
-	        return true;
+
+    //to execute SAST & SCA scans consecutively, 1st perform method call to execute SAST then after changing the scan type to SCA
```

Comment:
Rather than having all of these checks for SCA and SAST and rather than trying to execute the perform() method twice, we should create a new scanner type in the SDK that will execute both a SAST scan and an SCA scan. That would eliminate the need for most, if not all, of these changes.

Reply:
SAST_SCA class had been made to handle the SAST+SCA scans execution.